src/main/java/frc/robot/hardware/empties/EmptyDevice.java

```diff
@@ -0,0 +1,27 @@
+package frc.robot.hardware.empties;
+
+import frc.robot.hardware.ConnectedInputAutoLogged;
+import frc.robot.hardware.interfaces.IDevice;
+import frc.robot.hardware.interfaces.InputSignal;
+import org.littletonrobotics.junction.Logger;
+
+public abstract class EmptyDevice implements IDevice {
+
+	private final String logPath;
+	private final ConnectedInputAutoLogged connectedInput;
+
+	public EmptyDevice(String logPath) {
+		this.logPath = logPath;
+		this.connectedInput = new ConnectedInputAutoLogged();
```

Comment:
Breaking SOLID (DIP)

Reply:
What do you mean ?
If u are talking about the connected, it should be new with no params always in all classes